### ISSUE TITLE
Sometime running server with distributed configuration in 2.0.10/2.0.9 generate exception: java.lang.ClassCastException

### ISSUE THREAD

**Issue by shlapkov** [orientechnologies/orientdb#4381]
We are running our OrientDBCluster.java test but we have to change: NUM_WRITERS = 5; to NUM_WRITERS = 50; after this with Orient DB 2.0.10, 2.0.9 but not with 2.0.7 we are getting exception:

```
2015-06-17 15:36:10:820 SEVERE ******************************** [OrientDBCluster]
2015-06-17 15:36:10:821 SEVERE ******************************** [OrientDBCluster]Create Iteration=4, id=plocal:D:/SVN/dev/tools/orientdb-cluster-test/servers/0/databases/platform.46-37665e24-10d2-437b-9410-2b5dd9f5ef60, com.orientechnologies.orient.core.exception.OTransactionException: Error on committing distributed transaction
com.orientechnologies.orient.core.exception.OTransactionException: Error on committing distributed transaction
	at com.orientechnologies.orient.server.distributed.ODistributedStorage.commit(ODistributedStorage.java:953)
	at com.orientechnologies.orient.core.tx.OTransactionOptimistic.doCommit(OTransactionOptimistic.java:488)
	at com.orientechnologies.orient.core.tx.OTransactionOptimistic.commit(OTransactionOptimistic.java:147)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.commit(ODatabaseDocumentTx.java:2427)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.commit(ODatabaseDocumentTx.java:2397)
	at test.OrientDBCluster$Writer.createRecord(OrientDBCluster.java:360)
	at test.OrientDBCluster$Writer.call(OrientDBCluster.java:205)
	at test.OrientDBCluster$Writer.call(OrientDBCluster.java:187)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.orientechnologies.orient.core.exception.OStorageException: Error during transaction commit.
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.commit(OAbstractPaginatedStorage.java:941)
	at com.orientechnologies.orient.server.distributed.ODistributedStorage.commit(ODistributedStorage.java:816)
	at com.orientechnologies.orient.core.tx.OTransactionOptimistic.doCommit(OTransactionOptimistic.java:488)
	at com.orientechnologies.orient.core.tx.OTransactionOptimistic.commit(OTransactionOptimistic.java:147)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.commit(ODatabaseDocumentTx.java:2427)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.commit(ODatabaseDocumentTx.java:2397)
	at com.orientechnologies.orient.server.distributed.task.OTxTask.execute(OTxTask.java:116)
	at com.orientechnologies.orient.server.hazelcast.OHazelcastPlugin.executeOnLocalNode(OHazelcastPlugin.java:761)
	at com.orientechnologies.orient.server.hazelcast.ODistributedWorker.onMessage(ODistributedWorker.java:300)
	at com.orientechnologies.orient.server.hazelcast.ODistributedWorker.run(ODistributedWorker.java:121)
Caused by: java.lang.ClassCastException: com.orientechnologies.orient.server.distributed.ODistributedStorage cannot be cast to com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage
	at com.orientechnologies.orient.core.serialization.serializer.stream.OStreamSerializerSBTreeIndexRIDContainer.deserializeFromDirectMemoryObject(OStreamSerializerSBTreeIndexRIDContainer.java:258)
	at com.orientechnologies.orient.core.serialization.serializer.stream.OStreamSerializerSBTreeIndexRIDContainer.deserializeFromDirectMemoryObject(OStreamSerializerSBTreeIndexRIDContainer.java:40)
	at com.orientechnologies.orient.core.storage.impl.local.paginated.base.ODurablePage.deserializeFromDirectMemory(ODurablePage.java:129)
	at com.orientechnologies.orient.core.index.sbtree.local.OSBTreeBucket.getEntry(OSBTreeBucket.java:215)
	at com.orientechnologies.orient.core.index.sbtree.local.OSBTree.get(OSBTree.java:203)
	at com.orientechnologies.orient.core.index.engine.OSBTreeIndexEngine.get(OSBTreeIndexEngine.java:225)
	at com.orientechnologies.orient.core.index.OIndexMultiValues.putInSnapshot(OIndexMultiValues.java:162)
	at com.orientechnologies.orient.core.index.OIndexAbstract.applyIndexTxEntry(OIndexAbstract.java:1039)
	at com.orientechnologies.orient.core.index.OIndexAbstract.addTxOperation(OIndexAbstract.java:686)
	at com.orientechnologies.orient.core.tx.OTransactionOptimistic$CommitIndexesCallback.run(OTransactionOptimistic.java:98)
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.commit(OAbstractPaginatedStorage.java:927)
	... 9 more

2015-06-17 15:36:10:823 SEVERE ******************************** [OrientDBCluster]
2015-06-17 15:36:10:857 SEVERE ******************************** [OrientDBCluster]Update Iteration=0, id=plocal:D:/SVN/dev/tools/orientdb-cluster-test/servers/0/databases/platform.18-124c4277-1532-4345-8198-fb338ef064b9, java.lang.IllegalStateException: Cannot find record with id=plocal:D:/SVN/dev/tools/orientdb-cluster-test/servers/0/databases/platform.18-124c4277-1532-4345-8198-fb338ef064b9
java.lang.IllegalStateException: Cannot find record with id=plocal:D:/SVN/dev/tools/orientdb-cluster-test/servers/0/databases/platform.18-124c4277-1532-4345-8198-fb338ef064b9
	at test.OrientDBCluster$Writer.findById(OrientDBCluster.java:509)
	at test.OrientDBCluster$Writer.updateRecord(OrientDBCluster.java:261)
	at test.OrientDBCluster$Writer.call(OrientDBCluster.java:210)
	at test.OrientDBCluster$Writer.call(OrientDBCluster.java:187)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745) 
```

**Comment by lvca** [comment]
This issue has been already fixed in 2.0.11-SNAPSHOT. Please could you try with it? If everything is ok we could release 2.0.11 tomorrow.

**Comment by shlapkov** [comment]
Yes with 2.0.11-SNAPSHOT working fine. Thank you for fast response! Can we have fix asap?

**Comment by lvca** [comment]
Yes, we will release 2.0.11 tomorrow.

**Comment by lvca** [comment]
Hey @shlapkov, we've released 2.0.11.

**Comment by shlapkov** [comment]
yes @lvca thank you for update!
